nuimo/src/main/kotlin/com/senic/nuimo/NuimoDiscoveryManager.kt

```diff
@@ -205,10 +213,10 @@ class NuimoDiscoveryManager(context: Context) {
     private fun onDeviceFound(device: BluetoothDevice) {
         when {
             !nuimoDeviceNames.contains(device.name)                    -> return
-            discoveredControllers.any { it.address == device.address } -> return
+            discoveredControllers.any { it.nuimoController.address == device.address && it.updateDiscoveryTimestamp() } -> return
```

Comment:
I'd make this explicit to avoid the side effect within the functional code:

var knownController = discoveredControllers.first { it.nuimoController.address == device.address } // Or however `first` is expressed in Kotlin
if (knownController != null) {
  knownController.updateDiscoveryTimestamp()
  return
}